### DISCUSSION TITLE
The trailingSlash configuration is adding trailingSlash to alternates.canonical when the url have query strings

### DISCUSSION THREAD

**Discussion by HelaGone** [vercel/next.js#65304] [Help]
### Summary

For SEO reasons I need to declare the canonical url with the query strings, but the trailingSlash is being added to the canonical url. `https://exaple.com/discover/?id=123456-qwerty/` which is incorrect. At the same time I can't set to false the configuration for the trailingSlash or Google will deindex all my content from its index since that change will result in a lot of 308 redirects. How to prevent this behavior? This is happening in version 14.1.1

**Comment by JurajKavka** [comment]
I don't understand what `trailingSlash` option is supposed to do... it should handle redirects not manipulate with URLs. For example I can't make `<Link>` that contains hash `#` in URL. After click I end up with hash URL and trailing slash and the end i.e. `http://localhost:3000/about/#somehashparam/` I automatically generates URLs in `href=""` attr with trailing slash event I have just hash param at the end...

It is definitely BUG, because according to docs, It should handle redirects not to generate URLs with/without trailing slash in HTML elements https://nextjs.org/docs/app/api-reference/next-config-js/trailingSlash

**Reply by JurajKavka** [reply]
And It should redirect just `pathname` not path with query or hash params
